cunumeric/array.py

```diff
@@ -2465,6 +2465,58 @@ def diagonal(
                 raise ValueError("Either axis1/axis2 or axes must be supplied")
         return self._diag_helper(offset=offset, axes=axes, extract=extract)
 
+    @add_boilerplate("indices", "values")
+    def put(
+        self, indices: ndarray, values: ndarray, mode: str = "raise"
+    ) -> None:
+        """
+        Set storage-indexed locations to corresponding values.
+
+        See Also
+        --------
+         numpy.put
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+
+        """
+
+        if values.size == 0 or indices.size == 0:
+            return
+
+        if mode not in ("raise", "wrap", "clip"):
+            raise ValueError("clipmode not understood")
+        if mode == "wrap":
+            indices = indices % self.size
+        elif mode == "clip":
+            indices = indices.clip(0, self.size - 1)
+
+        if indices.dtype != np.int64:
+            runtime.warn(
+                "converting index array to int64 type",
+                category=RuntimeWarning,
+            )
+            indices = indices.astype(np.int64)
```

Comment:
I feel like this happens somewhat frequently. Is it worth having a *"check dtype and covert with a warning if necessary"* helper function?